cfpb/consumerfinance.gov
# Add `o-search-input` component

## Additions

- Add new search input component pattern.

## Changes

- Migrate various search inputs to single search input component.


## How to test this PR

The (x) inside the search input should not show up unless text is entered. It should darken on hover and it should get a focus rectangle on tabbing to the (x) icon. 

Global search - Any page expand the global search.

Ask landing - search should show autocomplete
http://localhost:8000/ask-cfpb/

Ask landing Spanish - search show autocomplete
http://localhost:8000/es/obtener-respuestas/

Ask search - search should not show autocomplete
http://localhost:8000/ask-cfpb/search

eRegs
http://localhost:8000/rules-policy/regulations/search-regulations/results/?regs=1002

TDP
http://localhost:8000/consumer-tools/educator-tools/youth-financial-education/teach/activities/

Prepaid
http://localhost:8000/data-research/prepaid-accounts/search-agreements/


## Screenshots

Ask CFPB:


https://github.com/cfpb/consumerfinance.gov/assets/704760/10e2e2fe-acfc-415a-a3d1-878bc3bfd51b


Before (mobile view):
<img width="519" alt="Screenshot 2024-05-10 at 1 59 01 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/60d3cb8a-410f-46b1-84d0-4d3f365dd225">

After (mobile view):
<img width="546" alt="Screenshot 2024-05-10 at 1 58 33 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/f43362ba-a3d4-495e-a125-a12d2dc9692a">


Regs:

Before:
<img width="1063" alt="Screenshot 2024-05-10 at 2 10 50 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/1de245d0-40d1-47a6-b1b8-8f857a12b738">

After:
<img width="1144" alt="Screenshot 2024-05-10 at 2 10 40 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/57fcc41a-01ad-4dcc-b530-2ef836b1ac16">


TDP:

Before:
<img width="1249" alt="Screenshot 2024-05-10 at 2 12 12 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/8447b36b-0510-4b1a-a6da-212567c4fd72">

After:
<img width="1231" alt="Screenshot 2024-05-10 at 2 12 02 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/16fb989a-f054-41d1-8c6c-43f8ead4c54c">


Prepaid:

Before:
<img width="1217" alt="Screenshot 2024-05-10 at 2 13 23 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/6a370b16-a9b4-4ad5-aa4c-6b2a854b4b60">

After:
<img width="1246" alt="Screenshot 2024-05-10 at 2 13 10 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/09570767-8180-4d77-9625-4a7d3006b8fe">


iOS 17.2

<img width="420" alt="Screenshot 2024-05-10 at 5 21 29 PM" src="https://github.com/cfpb/consumerfinance.gov/assets/704760/9bb27107-4957-4ae2-8142-935ced1c2450">

## Note

- This pattern will be added to the DS, once this is approved.
- The Ask answers  page should maybe use this component too, see version on
http://localhost:8000/consumer-tools/auto-loans/answers/
- The touch target makes the button wider than the x icon, but to align it correctly the icon is offset from the button center. To make the focus rectangle look correct, it is only on the icon. This means it's possible to push the button to the left of the focus rectangle, but I think this is fine.
